### PR TITLE
Chapter 2: Android and Model-View-Controller

### PR DESCRIPTION
**Pull request by OnicaAdrianCatalin**
Screenshot 2022-04-07 at 13 40 28

I applied MVC pattern in code.

### CODE REVIEW

File: app/src/main/res/drawable/ic_arrow_left.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (NegreaVlad) [inline app/src/main/res/drawable/ic_arrow_left.xml:1]: I like that you used vector images

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+the Atlantic Ocean.</string>
+        <string name="question_mideast">The Suez Canal connects the Red Sea
+and the Indian Ocean.</string>
+        <string name="question_africa">The source of the Nile River is in Egypt.</string>
+        <string name="question_americas">The Amazon River is the longest river
+in the Americas.</string>
+        <string name="question_asia">Lake Baikal is the world\'s oldest and deepest
+freshwater lake.</string>
+        <string name="previous_button_image">True</string>
+        <string name="next_button_image">True</string>
```
Comment (NegreaVlad) [inline app/src/main/res/values/strings.xml]: The string names should represent the values, not the usages.
There should be a single string with value "True" that's shared across all usages.

Refactoring example:
`<string name="true">True</string>`

**Review by NegreaVlad — COMMENTED** [review]
I saw some spacing issues. Those will be resolved when we will include detekt in the project.
I will make a presentation on that.

Some issues with the string resources, other than that, everything is good.

It would be very helpful to include a screenshot of the app to better understand the result.
You can take screenshots from AS: https://developer.android.com/studio/debug/am-screenshot

File: app/src/main/java/com/example/geoquiz/MainActivity.kt [hunk 3]
```diff
@@ -5,30 +5,71 @@ import android.os.Bundle
...
         trueButton = findViewById(R.id.true_button)
         falseButton = findViewById(R.id.false_button)
+        nextButton = findViewById(R.id.next_button)
+        previousButton = findViewById(R.id.previous_button)
+        questionTextView = findViewById(R.id.question_text_view)
+
+
 
         trueButton.setOnClickListener {
-            // CHALLENGE: Customize the toast to show at the top instead of the bottom of the screen
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/MainActivity.kt:22]: One small suggestion for later on when things will become crowded in the onCreate method. I like to create a separate `private fun setOnClickListeners()` function where I do all my on click listeners setup. This way I keep it nice and organised no matter how much code is added.

**Review by iugaliviu — COMMENTED** [review]
Apart from what Vlad already mentioned regarding strings and styling it looks good to me. Good job!

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
-    <string name="correct_toast">Correct!</string>
-    <string name="incorrect_toast">Incorrect</string>
+        <string name="app_name">Test</string>
+        <string name="true_button">True</string>
+        <string name="false_button">False</string>
+        <string name="question_australia">Canberra is the capital of Australia.</string>
+        <string name="correct">Correct!</string>
+        <string name="incorrect">Incorrect!</string>
+        <string name="next_button">Next</string>
+        <string name="previous_button">Previous</string>
```
Comment (NegreaVlad) [inline app/src/main/res/values/strings.xml]: remove the _button suffix and you will see that you have this string duplicated. Use just one in both places.
Remove the _button suffix in all cases

**Review by NegreaVlad — COMMENTED** [review]
Just some small changes required in strings naming. Good work!

**Review by NegreaVlad — APPROVED** [review]
Looks good to me (LGTM)! Thank you!

**Pull request merged**